system-parachains/asset-hubs/asset-hub-polkadot/src/staking/mod.rs

```diff
@@ -252,6 +254,140 @@ impl multi_block::unsigned::miner::MinerConfig for Runtime {
 	type TargetSnapshotPerBlock = <Runtime as multi_block::Config>::TargetSnapshotPerBlock;
 }
 
+// Wittled copy of the code from https://github.com/paritytech/polkadot-sdk/pull/9556
+// To be replaced after that is merged and available.
+pub mod temp_curve {
+	use super::*;
+	use sp_runtime::traits::{AtLeast32BitUnsigned, One};
+	use scale_info::TypeInfo;
+	use sp_arithmetic::traits::Saturating;
+
+	/// The step type for the stepped curve.
+	#[derive(PartialEq, Eq, sp_core::RuntimeDebug, TypeInfo, Clone)]
+	pub enum Step<V> {
+		/// Move towards a desired value by a percentage of the remaining difference at each step.
+		///
+		/// Step size will be (target_total - current_value) * pct.
+		RemainingPct(V, Perbill),
+	}
+
+	/// A stepped curve.
+	///
+	/// Steps every `period` from the `initial_value` as defined by `step`.
+	/// First step from `initial_value` takes place at `start` + `period`.
+	#[derive(PartialEq, Eq, sp_core::RuntimeDebug, TypeInfo, Clone)]
+	pub struct SteppedCurve<P, V> {
```

Comment:
```suggestion
	pub struct SteppedCurve<X, Y> {
```
Maybe this would help with the terminology to be a bit more mathematical.